apps/rpc/src/modules/event/event-router.ts

```diff
@@ -89,28 +83,21 @@ export const eventRouter = t.router({
     }),
 
   all: procedure
-    .input(
-      z
-        .object({
-          page: PaginateInputSchema,
-          filter: EventFilterQuerySchema.optional(),
-        })
-        .optional()
-    )
-    .output(z.array(EventSchema))
+    .input(BasePaginateInputSchema.extend({ filter: EventFilterQuerySchema.optional() }).optional())
+    .query(async ({ input, ctx }) =>
+      ctx.executeTransaction(async (handle) => await ctx.eventService.findEvents(handle, { ...input?.filter }, input))
+    ),
+
+  allPaged: procedure
```

Comment:
I'm a little confused, why do we need both all and allPages? These should be one query